cmd/wait-for-github/ci.go

```diff
@@ -74,108 +74,127 @@ func parseCIArguments(c *cli.Context) error {
 		cli.ShowCommandHelpAndExit(parent, "ci", 1)
 
 		// shouldn't get here, the previous line should exit
-		return nil
+		return ciConfig{}, nil
 	}
 
-	ciConf.owner = owner
-	ciConf.repo = repo
-	ciConf.ref = ref
-	ciConf.checks = c.StringSlice("check")
-
-	log.Debugf("Will wait for CI on %s/%s@%s", owner, repo, ref)
-
-	return nil
+	return ciConfig{
+		owner:  owner,
+		repo:   repo,
+		ref:    ref,
+		checks: c.StringSlice("check"),
+	}, nil
 }
 
-func handleCIStatus(status github.CIStatus) cli.ExitCoder {
+func handleCIStatus(status github.CIStatus, recheckInterval time.Duration) cli.ExitCoder {
 	switch status {
 	case github.CIStatusUnknown:
-		log.Infof("CI status is unknown, rechecking in %s", cfg.recheckInterval)
+		log.Infof("CI status is unknown, rechecking in %s", recheckInterval)
 	case github.CIStatusPending:
 	case github.CIStatusPassed:
 		return cli.Exit("CI successful", 0)
 	case github.CIStatusFailed:
 		return cli.Exit("CI failed", 1)
 	}
 
-	log.Infof("CI is not finished yet, rechecking in %s", cfg.recheckInterval)
+	log.Infof("CI is not finished yet, rechecking in %s", recheckInterval)
 	return nil
 }
 
-func checkCIStatus(c *cli.Context) error {
-	ctx := context.Background()
+type checkSpecificCI struct {
```

Comment:
`checkSpecificCI`. Not sure about the name. What part is specific?

Reply:
It's the `checks` field. When we use this one we call `GetCIStatusForChecks`, it takes a slice of check/status names to look for and makes sure they are all there.

However you made me realise that the structs could be a bit better if the `checks` slice wasn't in the `checkAllCI` one, so I [reorganised them and re-pushed](https://github.com/grafana/wait-for-github/compare/e92072b83a7ff3a36a70bb4b950f944782b27d63..781f8e64edb1ba89fb1fb7bf9dec0939525a967a). Thanks for that :+1: